### DISCUSSION TITLE
integrations/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17926] [Docs]
# integrations/

Discover Ultralytics integrations for streamlined ML workflows, dataset management, optimized model training, and robust deployment solutions.

https://docs.ultralytics.com/integrations/

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! Our integrations are designed to help streamline your ML workflows, from dataset management to optimized model training and deployment. You can explore all the details in our [Integrations Documentation](https://docs.ultralytics.com/integrations/) for step-by-step guidance.

If this is a 🐛 Bug Report, kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in debugging the issue.

If you have a custom ❓ Question about integrations, please provide specific details on what you're trying to achieve, including environment details, code snippets, or relevant logs. This will help us better understand your goals or issues.

We're here to help, and an Ultralytics engineer will follow up with your discussion soon. In the meantime, feel free to connect with the Ultralytics community for more insights:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧  
- In-depth discussions: [Discourse](https://community.ultralytics.com)  
- Reddit threads: [Subreddit](https://reddit.com/r/Ultralytics)  

## Upgrade

If you haven’t already, ensure you're using the latest version of the `ultralytics` package along with the necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Our tools are tested and verified in the following environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Ensure to check the status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests, which validate various modes and tasks across platforms.

Stay tuned for further guidance, and thank you for exploring Ultralytics integrations! 😊
